docs/azure/sdk/snippets/authentication/user-assigned-managed-identity/Program.cs

```diff
@@ -0,0 +1,190 @@
+﻿using Azure.Identity;
+using Microsoft.Extensions.Azure;
+using Azure.Storage.Blobs;
+using Azure.Core;
+
+var userAssignedClientId = "<user-assigned-client-id>";
+var builder = WebApplication.CreateBuilder(args);
+
+// Registration options
+// registerUsingClientId(builder);
+// registerUsingObjectId(builder);
+// registerUsingResourceId(builder);
+
+var app = builder.Build();
+
+if (app.Environment.IsDevelopment())
+{
+    app.UseSwagger();
+    app.UseSwaggerUI();
+}
+
+app.UseHttpsRedirection();
+
+var summaries = new[]
+{
+    "Freezing", "Bracing", "Chilly", "Cool", "Mild", "Warm", "Balmy", "Hot", "Sweltering", "Scorching"
+};
+
+app.MapGet("/weatherforecast", () =>
+{
+    var forecast = Enumerable.Range(1, 5).Select(index =>
+        new WeatherForecast
+        (
+            DateOnly.FromDateTime(DateTime.Now.AddDays(index)),
+            Random.Shared.Next(-20, 55),
+            summaries[Random.Shared.Next(summaries.Length)]
+        ))
+        .ToArray();
+    return forecast;
+})
+.WithName("GetWeatherForecast")
+.WithOpenApi();
+
+app.Run();
+
+void registerUsingClientId(WebApplicationBuilder builder)
+{
+    #region snippet_MIC_ClientId_UseCredential
+    builder.Services.AddAzureClients(clientBuilder =>
+    {
+        clientBuilder.AddBlobServiceClient(
+            new Uri("https://<account-name>.blob.core.windows.net"));
+
+        TokenCredential credential = null;
+
+        if (builder.Environment.IsProduction() || builder.Environment.IsStaging())
+        {
+            // Managed identity token credential discovered when running in Azure environments
+            credential = new ManagedIdentityCredential(
+                ManagedIdentityId.FromUserAssignedClientId("<client-id>"));
```

Comment:
```suggestion
                ManagedIdentityId.FromUserAssignedClientId(userAssignedClientId ));
```

Reply:
Let's keep the and remove the unused variable so that each code section stands on its own.